src/s25update.cpp

```diff
@@ -114,7 +114,11 @@ size_t WriteMemoryCallback(void* ptr, size_t size, size_t nmemb, std::string* da
 /**
  *  curl progressbar callback
  */
+#if LIBCURL_VERSION_NUM >= 0x011DF0
```

Comment:
Is that number correct? https://curl.se/docs/versions.html says `#define LIBCURL_VERSION_NUM 0xXXYYZZ`  so it should rather be `0x073200` shouldn't it? Or `0x072000` ~I even see https://github.com/curl/curl/blob/master/include/curl/curlver.h#L76 which means you can use `#if CURL_AT_LEAST_VERSION(7,32,0)` which is a lot more readable~ but only available in curl 7.43+ but we can copy the macro definition when it is not defined and use it if it is.

Reply:
Uff. my bad... shouldn't have just copied the output of "dec to hex converter" for 073200. My bad.
correct version string is now defined.

Will also comment on the conversion like in curl/curlver.h which comment about it in greater detail. For my bad it also did just work on my machine. One Moment.

Reply:
How about adding this:
```
#ifndef CURL_AT_LEAST_VERSION
#define CURL_AT_LEAST_VERSION(x,y,z) \
  (LIBCURL_VERSION_NUM >= ((x)<<16|(y)<<8|(z)))
#endif
```

Reply:
Yes, this is useful and readable. Where to put best? The curl.h is included in the s25update.cpp file. So probably best to put it there, correct?

Reply:
Yes exactly. Right after the includes I guess is best. Note that you can use clang-format-10 to auto-format files.

Reply:
`/*
 * CURL_AT_LEAST_VERSION took from libCURL 7.43+ of curl/curlver.h for easier readability.
 */
`

Is now my comment regarding that.